docker/src/ope-notebook-culler.sh

```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
+
+running_notebooks=$(oc get notebook -o custom-columns=:metadata.name)
+
+echo "Running ope-notebook-culler.sh..."
+echo $(date)
+
+for notebook in $running_notebooks; do
+	oc delete notebook.kubeflow.org $notebook
```

Comment:
I think this comment won't delete the pvcs connected to the notebooks. When the same student relaunch a notebook, will that pvc be automatically assigned to the user?

Reply:
Now I think about it, `oc delete` command will equally kill all notebooks no matter how long they have been running. One user may just start a notebook 1 minute before. Then the cronjob kills it immediately. That's not what we expect. Can you make the cronjob job run every 24 hours, and then only kill notebooks that are running more than 48 hours?

Reply:
Both of these assumptions are correct, the pvc does need to be deleted and we don't want to delete notebooks that have been running less than 48 hours. The script I had before checked the running time of the notebook and deleted the pvc and notebook, maybe that can be used, I will copy it below:

`       
        #!/bin/sh
        
        set_time=24 hours

        time_limit=$(($(date -u +"%s") - $(date -d '-$set_time' +%s)))

        running_notebooks=$(oc get notebook -o custom-columns=:metadata.name)

        for notebook in $running_notebooks; do

          start_time=$(oc get notebook $notebook -o jsonpath='{.status.containerState.running.startedAt}')
            
          running_time=$(($(date -u +"%s") - $(date -u -d "$start_time" +"%s")))
          
          if [ "$running_time" -gt "$time_limit" ]; then
            echo "Shutting down pod $notebook"
            oc delete notebook $notebook
            oc delete pvc $notebook
          fi 
        done
`

Reply:
@dystewart is it necessary to delete the whole notebook, or can the StatefulSet be scaled to 0 instead?

Reply:
@computate If I'm not mistaken, when you scale the StatefulSet to 0 the pod is terminated but then starts back up again. That's why we delete the whole notebook because it ensures the notebook, statefulset and pod are actually deleted.